modules/Registration/containers/student/StudentRegistrationContainer.tsx

```diff
@@ -49,10 +50,10 @@ export function StudentRegistrationContainer() {
         router.push("/dashboard");
       }, 100);
     } catch (error) {
-      console.error("Registration failed:", error);
+      const apiError = error as APIError;
       notifications.show({
         title: "Error",
-        message: "Registration failed. Please try again.",
+        message: apiError.data.message,
         color: "red",
```

Comment:
You probably need this for multiple forms throughout the app.
Can we can abstract this into a shared helper method which takes the error as a param?